components/requests/modals/EditDoctorInformationModal.tsx

```diff
@@ -18,58 +18,75 @@ import {
   useToast,
 } from '@chakra-ui/react'; // Chakra UI
 import SuccessfulEditAlert from '@components/permit-holders/SuccessfulEditAlert'; // Successful edit alert/toast
-import { useState, useEffect, SyntheticEvent, ReactNode } from 'react'; // React
+import { useState, SyntheticEvent, ReactNode } from 'react'; // React
 import { DoctorInformationCardPhysician } from '@tools/components/internal/requests/doctor-information-card'; // Physician type
+import { useMutation } from '@apollo/client'; // Apollo Client
+import {
+  UPSERT_PHYSICIAN_MUTATION,
+  UpsertPhysicianRequest,
+  UpsertPhysicianResponse,
+} from '@tools/pages/admin/permit-holders/upsert-physician'; // Page tools
 
 type EditDoctorInformationModalProps = {
   children: ReactNode;
   readonly physician: DoctorInformationCardPhysician;
   readonly onSave: (applicationData: any) => void;
 };
 
-export default function EditDoctorInformationModal({
-  children,
-  physician,
-  onSave,
-}: EditDoctorInformationModalProps) {
+export default function EditDoctorInformationModal({ children }: EditDoctorInformationModalProps) {
   const { isOpen, onOpen, onClose } = useDisclosure();
 
   const [name, setName] = useState('');
-  const [mspNumber, setMspNumber] = useState<number | undefined>();
+  const [mspNumber, setMspNumber] = useState<number>(0);
   const [phoneNumber, setPhoneNumber] = useState('');
   const [addressLine1, setAddressLine1] = useState('');
   const [addressLine2, setAddressLine2] = useState<string | undefined>('');
   const [city, setCity] = useState('');
   const [postalCode, setPostalCode] = useState('');
 
-  const successfulEditToast = useToast();
-  useEffect(() => {
-    setName(physician.name);
-    setMspNumber(physician.mspNumber);
-    setPhoneNumber(physician.phone);
-    setAddressLine1(physician.addressLine1);
-    setAddressLine2(physician.addressLine2 || undefined);
-    setCity(physician.city);
-    setPostalCode(physician.postalCode);
-  }, [physician]);
```

Comment:
Don't we need this `useEffect` here for the initial data?

Reply:
@chrischan325 please leave this in for now

Reply:
do I leave it in and comment it out or just leave it, there are some errors and if left uncommented but I can fix them

Reply:
i think @jeffreyzhang2001 wrote this part so he might have more context